apis/rbac/helpers.go

```diff
@@ -0,0 +1,65 @@
+package rbac
+
+// APIGroupMatches returns whether the specified requestedAPIGroup matches any of
+// the rule API groups
+func (r Rule) APIGroupMatches(requestedAPIGroup string) bool {
```

Comment:
Could this interface be simplified to:

```golang
func (r Rule) Allows(obj meta.Object, verb string) bool {}
```

You can get the GroupVersionKind with `meta.GetGroupVersionKind(obj)` and and the name is accessible via `.GetName()`.

Reply:
We don't necessarily have an object here; `get` or `list` requests for example wouldn't provide an object in the request body.

The request attributes (here `requestedAPIGroup`) would be extracted from the request URI by a function similar to https://github.com/kubernetes/apiserver/blob/45bb707b3e17de3fa72c53d6df5403fea2c3150a/pkg/endpoints/request/requestinfo.go#L116.

Reply:
You're right. We don't necessarily have an object yet at all.